src/createSlice.ts

```diff
@@ -81,7 +89,11 @@ export interface CreateSliceOptions<
    * functions. For every action type, a matching action creator will be
    * generated using `createAction()`.
    */
-  reducers: ValidateSliceCaseReducers<State, CR>
+  reducers:
+    | ValidateSliceCaseReducers<State, CR>
+    | ((
+        creators: ReducerCreators<State>
+      ) => CR) /* & ValidateSliceCaseReducers<State, CR> */ // TODO: possible?
```

Comment:
If this is not possible (which I'm not sure about), it would not be possible to test the result of the builder notation if the `payload` of reducers created with the `prepare` notation would be correct between the reducer & the prepare function.

=> Maybe we should forbid that definition here and add another `withSpecial.reducer` function for usage with the builder notation? Would be more "in theme" anyways.

Reply:
I've added `create.preparedReducer` as a new method and throw an error if the `prepare` notation is used without that in combination with the `create` notation, as in that case we cannot correctly check if the payload returned from `prepare` is the same payload as expected by the reducer.
It's unfortunate, but when I try, the inference breaks in all directions due to the type being inferred from a method return value in the first place.

On the other hand, it's probably more consistent anyways.